### PR TITLE
addition of private endpoint

### PR DESCRIPTION
**Pull request by dgcaron**
Objective:

Add support for azurerm_private_endpoint.

Example output without usage-file:
```
azurerm_private_endpoint.without_usage_file                                                             
 ├─ Private endpoint                                         730  hour                             $7.30 
 ├─ Inbound data processed                    Monthly cost depends on usage: $0.01 per GB                
 └─ Outbound data processed                 Monthly cost depends on usage: $0.01 per GB                
```

Example output with usage-file:
```
  azurerm_private_endpoint.with_inbound                                                                   
 ├─ Private endpoint                                         730  hour                             $7.30 
 ├─ Inbound data processed                                   100  GB                               $1.00 
 └─ Outbound data processed                   Monthly cost depends on usage: $0.01 per GB                
                                                                                                         
 azurerm_private_endpoint.with_outbound                                                                  
 ├─ Private endpoint                                         730  hour                             $7.30 
 ├─ Inbound data processed                    Monthly cost depends on usage: $0.01 per GB                
 └─ Outbound data processed                                  100  GB                               $1.00 
```

Pricing details:
```
	db.products.find({"vendorName": "azure", "service": "Virtual Network", "productFamily": "Networking", "attributes.productName":"Virtual Network Private Link", "attributes.meterName": "Private Endpoint" }).pretty()
	db.products.find({"vendorName": "azure", "service": "Virtual Network", "productFamily": "Networking", "attributes.productName":"Virtual Network Private Link", "attributes.meterName": "Data Processed - Egress" }).pretty()
	db.products.find({"vendorName": "azure", "service": "Virtual Network", "productFamily": "Networking", "attributes.productName":"Virtual Network Private Link", "attributes.meterName": "Data Processed - Ingress" }).pretty()
```

Status:

[x] Added to azure/registry.go
[ ] Added resource file
[x] Added integration tests
 Added unit tests if applicable - Not applicable
Issues:

None
Useful links:

Pricing: https://azure.microsoft.com/en-us/pricing/calculator -> search for Azure Private Link
Terraform: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/private_endpoint

### CODE REVIEW

**Comment by alikhajeh1** [comment]
@dgcaron thanks for the PR! I'm fixing the tests in master (they started failing due to [this](https://github.com/hashicorp/terraform-provider-aws/pull/10807#issuecomment-885713337))

**Comment by dgcaron** [comment]
Ok, is there anything I need to add to this PR? Or correct?

**Review by alikhajeh1 — COMMENTED** [review]
This is an awesome PR! 

I added a few minor things.

https://github.com/infracost/infracost/pull/878#issuecomment-885710953 has a link at the bottom that could be followed to fix the issue I think
https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user

File: internal/providers/terraform/azure/private_endpoint.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	}
+}
+
+func privateEndpointCostComponent(region, name, meterName string) *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:                 name,
+		Unit:                 "hour",
+		UnitMultiplier:       1,
+		MonthlyQuantity:      decimalPtr(decimal.NewFromInt(730)),
+		IgnoreIfMissingPrice: true,
```
[the remark this answers is not in the document]
Comment (dgcaron) [reply internal/providers/terraform/azure/private_endpoint.go]: i have removed the two lines from the private_endpoint.go file

File: internal/providers/terraform/azure/testdata/private_endpoint_test/private_endpoint_test.usage.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+version: 0.1
```
[the remark this answers is not in the document]
Comment (dgcaron) [reply internal/providers/terraform/azure/testdata/private_endpoint_test/private_endpoint_test.usage.yml:1]: these lines have been added to the file

**Comment by alikhajeh1** [comment]
Thanks @dgcaron! This'll get released next week probably

**Comment by dgcaron** [comment]
this is great! i noticed the build was failing because the linter didn't run properly on my last commit, so I was fixing it but you already merged it. not sure if you were aware..

**Comment by alikhajeh1** [comment]
@dgcaron yep, https://github.com/infracost/infracost/commit/a0210fe8f5b989e66690f16feb7f683d1a2eddaf will fix it, I noticed it after I merged since the master branch tests were broken - that TF issue was more annoying than I thought 😅

**Pull request merged**